patches/tModLoader/Terraria/ModLoader/Default/ModAccessorySlotPlayer.cs

```diff
@@ -193,16 +285,242 @@ public override void SendClientChanges(ModPlayer clientPlayer)
+	/// </summary>
+	internal sealed class ExEquipmentLoadout
+	{
+		private readonly string identifier;
+
+		public ExEquipmentLoadout(int loadoutIndex, int slotCount, EquipmentLoadout loadoutReference)
+		{
+			this.LoadoutIndex = loadoutIndex;
+			this.LoadoutReference = loadoutReference;
+			this.identifier = $"loadout_{loadoutIndex}";
+
+			this.ResetAndSizeAccessoryArrays(slotCount);
+		}
+
+		public int LoadoutIndex { get; }
+
+		public EquipmentLoadout LoadoutReference { get; }
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exAccessorySlot"/>
+		public Item[] ExAccessorySlot { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exDyesAccessory"/>
+		public Item[] ExDyesAccessory { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exHideAccessory"/>
+		public bool[] ExHideAccessory { get; private set; } = [];
+
+		public void SaveData(TagCompound tag)
+		{
+			tag[identifier] = new TagCompound {
+				["items"] = ExAccessorySlot.Select(ItemIO.Save).ToList(),
+				["dyes"] = ExDyesAccessory.Select(ItemIO.Save).ToList(),
+				["hidden"] = ExHideAccessory.ToList(),
+			};
+		}
+
+		/// <summary>
+		/// Loads data for this loadout and updates this instance accordingly.
+		/// Returns a collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </summary>
+		/// <param name="tag">The <see cref="TagCompound"/> from which to load the data</param>
+		/// <param name="order">Saved slot names in order.</param>
+		/// <param name="slots">Slot name to slot info mapping.</param>
+		/// <returns>
+		/// A collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </returns>
+		public IReadOnlyList<SlotInfo> LoadData(
+			TagCompound tag,
+			List<string> order,
+			Dictionary<string, (int SlotType, bool HasLoadoutSupport)> slots)
+		{
+			List<SlotInfo> result = [];
+			IList<Item> items;
+			IList<Item> dyes;
+			IList<bool> visible;
+
+			this.ResetAndSizeAccessoryArrays(slots.Count);
+
+			tag = tag.GetCompound(identifier);
+			items = tag.GetList<TagCompound>("items").Select(ItemIO.Load).ToList();
+			dyes = tag.GetList<TagCompound>("dyes").Select(ItemIO.Load).ToList();
+			visible = tag.GetList<bool>("hidden").ToList();
+
+			for (int i = 0; i < order.Count; i++) {
+				(int type, bool hasLoadoutSupport) = slots[order[i]];
+
+				Item dye = dyes.ElementAtOrDefault(i) ?? new Item();
+				Item accessory = items.ElementAtOrDefault(i) ?? new Item();
+				Item vanityItem = items.ElementAtOrDefault(i + order.Count) ?? new Item();
+				bool isHidden = visible.ElementAtOrDefault(i);
```

Comment:
Why are we using `ElementAtOrDefault`? Is there some way the data could be missing? Scary